### PR TITLE
Tickets/dm 24264

### PR DESCRIPTION
**Pull request by patrickingraham**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -75,41 +78,73 @@ A general overview of the development flow, as viewed from the user, is as follo
...
+
+.. figure:: _static/notebook_and_queueJob_workflow_diagram.jpg
+    :width: 600px
+    :align: center
+    :alt: Workflow Summary
+
+    A visual overview of the workflow process, starting from an original idea first demonstrated in a Jupyter notebook.
+
+The two-tiered development strategy is being adopted to facilitate maximum flexibility to implement changes on
+short timescales. The staging (or development) area gives users full range to simultaneously operate across multiple
```
Comment (tribeiro) [inline index.rst]: The development and staging areas gives users full range to simultaneously operate across multiple

File: index.rst [hunk 2]
```diff
@@ -75,41 +78,73 @@ A general overview of the development flow, as viewed from the user, is as follo
...
+versions/branches with unhindered flexibility. This is especially important for commissioning activities where rapid
+code changes are required and temporary implementation is required which will break other people's code in the same
+area.
+
+The production area is to provide a single code base where all code in that area (on the master/develop branches) is
+fully functional at all times. This will be most important during operations when the codebase will be less dynamic
+but will also be useful for different commissioning teams that will circle through and want an established and bug-free
```
Comment (tribeiro) [inline index.rst]: Consider removing the "bug-free" part... I can see a lot of people rolling their eyes on the mention of bug-free software.

File: index.rst [hunk 3]
```diff
@@ -75,41 +78,73 @@ A general overview of the development flow, as viewed from the user, is as follo
...
+testing in order to ensure that changes made in one area do not result in breaking code in another.
+
+
 
 .. note::
 
-    This technote avoids the use of the term "scripts" and also assumes the current scriptQueue has been renamed. The
-    reasoning behind this is that "scripts" is an overloaded word where each person has their own interpretation where
-    the general tendency is to assume a bash-script type of execution.
+    This technote avoids the use of the term "scripts" and also assumes the current scriptQueue has been renamed to
```
Comment (tribeiro) [inline index.rst:110]: Are we still planning on doing this? Consider removing this comment. Or mentioned the ongoing ticket.

File: index.rst [hunk 4]
```diff
@@ -118,68 +153,81 @@ to diagnose or characterize certain behaviour), and personal notebooks. The deta
 scope of this technote. Until the re-organisation is completed, tags will be made of the repo at least every ~6 months,
```
Comment (tribeiro) [inline index.rst]: Simon made a valid point that removing files from a branch in a repo does not necessarily remove the file from the repo. I think we should rephrase this a bit. Maybe something on the lines of:

Until the re-organisation is completed, tags will be made of the repo at least every 6 months or before/after major activities. After each release, user will be asked to review and possibly remove notebooks older than 1 year to make sure stale notebooks are not lingering alongside the main working branch.

File: index.rst (hunk 2 above)
Comment (patrickingraham) [reply index.rst]: replaced with "stable"

File: index.rst (hunk 3 above)
Comment (patrickingraham) [reply index.rst:110]: Added reference to ticket and said document will be updated to reflect whatever is implemented.

File: index.rst (hunk 4 above)
Comment (patrickingraham) [reply index.rst]: Replaced

File: index.rst [hunk 5]
```diff
@@ -118,68 +153,81 @@ to diagnose or characterize certain behaviour), and personal notebooks. The deta
...
 after which all files larger than 20 MB (TBR) or older than 1 year will be deleted from the develop branch.
 
+.. note::
+
+    Another suggestion is to use git-lfs for large files in the repo. This needs further
```
Comment (tribeiro) [inline index.rst:161]: is the idea to use git-lfs for notebooks? I am not sure we will ever get notebooks that are large enough that will require that if we clean them up.

File: index.rst [hunk 6]
```diff
@@ -231,78 +279,148 @@ are captured correctly.
...
+
+- Validation of appropriate input types (dtypes)
+
+    - Verification of appropriate input values are only required if the values are not checked/verified elsewhere (such
+      as at lower levels (e.g. the CSCs).
+
+- Testing of end-to-end functionality for the primary functions for appropriate inputs
+
+    - E.g. does it correctly measure the centroid on a piece of test data to within a given tolerance?
+
```
Comment (tribeiro) [inline index.rst:318]: Maybe add:

- Testing that common edge cases are properly captured/treated. 

?

File: index.rst [hunk 7]
```diff
@@ -231,78 +279,148 @@ are captured correctly.
...
+- All jobs and utilities in the controls package must successfully pass all tests.
+
+    - Ideally this would be done automatically using a CI framework. If not available, then an artifact needs
+      to be shown as part of PR
+    - Tests have to pass **before merging** not just at the time of creating the PR.
+
+
+.. TODO::
+    DM has developed a way to do this, we should explore if this solution works for this case as well.
```
Comment (tribeiro) [inline index.rst:334]: I don't DM has actually done this. I think they have CI for package unit testing not integration testing.

**Review by tribeiro — APPROVED** [review]
I made some comments in line. Feel free to accept/reject them as suitable.

File: index.rst (hunk 6 above)
Comment (patrickingraham) [reply index.rst:318]: added

File: index.rst (hunk 7 above)
Comment (patrickingraham) [reply index.rst:334]: I thought they had.... which is what the comments just below that were about...

File: index.rst (hunk 5 above)
Comment (patrickingraham) [reply index.rst:161]: just left it as a suggestion.... i don't want to let this issue get in the way of us getting this document out the door...
